bluesky/preprocessors.py

```diff
@@ -289,6 +290,24 @@ def print_summary_wrapper(plan):
         cmd = msg.command
         if cmd == 'open_run':
             print('{:=^80}'.format(' Open Run '))
+            # This next bit checks the metadata for json compatibility
+            try:
+                json.dumps(msg.kwargs)
+            except TypeError:
+                error_keys=[]
+                for key, value in msg.kwargs.items():
+                    try:
+                        json.dumps(value)
+                    except TypeError:
+                        error_keys.append(key)
+                if error_keys:
+                    raise TypeError(
+                        f'The values for the keys in "{error_keys}" '
+                        f'from the start document are not valid')
+                else:
+                    raise TypeError(
```

Comment:
Is it possible to hit this codepath?

Reply:
Yes, the keys (not the values) could be what is failing to serialize.

Reply:
or the combination of 'keys' and 'values' could be failing to serailize (i.e it is not written as a valid dict !-) )